include/slurp.h

```diff
@@ -84,6 +84,11 @@ struct slurp_seat {
 	int32_t anchor_x, anchor_y;
 	struct slurp_box selection;
 	bool has_selection;
+
+	// touch:
+	struct wl_touch *wl_touch;
+	bool already_touched;
+	struct slurp_output *current_touch_output;
```

Comment:
Do we really need this field? Can't we just use `current_output`?

Reply:
Yes but this one is a bit trickier.

I've investigated further and found out that
1. pointer_leave event is triggered whenever touch down happens (even if mouse is not moved).
2. handle_pointer_leave clears the current_output. 

I think the solution is either 
 * checking if has_selection is true and then leaving current_output be
 * leaving pointer_handle_leave empty since clearing current_output is the only thing the method does (until multi-monitor funcitonality is added).

Which one is better?

Reply:
>pointer_leave event is triggered whenever touch down happens (even if mouse is not moved).

That sounds like a Sway bug.

Reply:
> That sounds like a Sway bug.

Confirmed as this doesn't happen in wayfire. Opened an issue https://github.com/swaywm/sway/issues/5145 for this one.

Reply:
Thanks for tracking it down!

Reply:
The "real" solution would be to handle pointer input and touch input completely separately. Basically put the currently shared state in a struct and have two separate fields with this struct (one for pointer, one for touch). It would be a bigger change though.

Reply:
Just to be sure I got it right:

```
//selection (pointer/touch):
	struct {
		struct slurp_output *current_output;
		struct slurp_output *current_touch_output;
		int32_t x, y;
		int32_t anchor_x, anchor_y;
		struct slurp_box selection;
		bool has_selection;
	} selection;
```

Something like this?

Reply:
The idea is to track the position of the pointer and touch points independently. So we'd have:

```
struct slurp_selection {
	struct slurp_output *current_output;
	int32_t x, y;
	int32_t anchor_x, anchor_y;
	struct slurp_box box;
	bool valid;
};

struct slurp_seat {
	…

	struct slurp_selection pointer;
	struct slurp_selection touch;
};
```

However, I'm fine with keeping this for a future pull request.

Now that the Sway patch has been merged, we can get rid of `current_touch_output`.

Reply:
And done 4e127e3386f9802c7976b07df5db2826efa36540.